src/Fixer/Phpdoc/AlignMultilineCommentFixer.php

```diff
@@ -0,0 +1,155 @@
+ * @author Julien Falque <julien.falque@gmail.com>
+ */
+final class AlignMultilineCommentFixer extends AbstractFixer implements ConfigurationDefinitionFixerInterface, WhitespacesAwareFixerInterface
+{
+    private $tokenKinds;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configure(array $configuration = null)
+    {
+        parent::configure($configuration);
+
+        $this->tokenKinds = [T_DOC_COMMENT];
+        if ('phpdocs_only' !== $this->configuration['comment_type']) {
+            $this->tokenKinds[] = T_COMMENT;
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Multiline doc comment: enforce asterisk start on each line [PSR-5] and align them.',
+            [
+                new CodeSample(
+'<?php
+    /**
+            * This is a DOC Comment
+with a line not prefixed with asterisk
+
+   */'
+                ),
+                new CodeSample(
+'<?php
+    /*
+            * This is a doc-like multiline comment
+*/',
+                    ['comment_type' => 'phpdocs_like']
+                ),
+                new CodeSample(
+'<?php
+    /*
+            * This is a doc-like multiline comment
+with a line not prefixed with asterisk
+
+   */',
+                    ['comment_type' => 'all_multiline']
+                ),
+            ]
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isCandidate(Tokens $tokens)
+    {
+        return $tokens->isAnyTokenKindsFound($this->tokenKinds);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function applyFix(\SplFileInfo $file, Tokens $tokens)
+    {
+        $lineEnding = $this->whitespacesConfig->getLineEnding();
+        foreach ($tokens as $index => $token) {
+            if (!$token->isGivenKind($this->tokenKinds)) {
+                continue;
+            }
+
+            $whitespace = '';
+            $previousIndex = $index - 1;
+            if ($tokens[$previousIndex]->isWhitespace()) {
+                $whitespace = $tokens[$previousIndex]->getContent();
+                --$previousIndex;
+            }
+            if ($tokens[$previousIndex]->isGivenKind(T_OPEN_TAG)) {
+                $whitespace = preg_replace('/\S/', '', $tokens[$previousIndex]->getContent()).$whitespace;
+            }
+
+            if (!preg_match('/\R([ \t]*)$/', $whitespace, $matches)) {
+                continue;
+            }
+
+            $indentation = $matches[1];
+            $lines = preg_split('/\R/', $token->getContent());
+            foreach ($lines as $lineNumber => $line) {
+                if (0 === $lineNumber) {
+                    continue;
+                }
+
+                $line = ltrim($line);
+                if ($token->isGivenKind(T_COMMENT) && (!isset($line[0]) || '*' !== $line[0])) {
+                    if ('all_multiline' !== $this->configuration['comment_type']) {
+                        continue 2;
```

Comment:
While I'm not against using `continue 2`, I think it can be replaced with something easier to read. As it's here to check whether all lines of the comment start with an `*`, maybe you could do the same with a regular expression before iterating the lines instead?